lib/Doctrine/DBAL/Driver/PDOStatementImplementations.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+namespace Doctrine\DBAL\Driver;
+
+use Doctrine\DBAL\Driver\PDO\Exception;
+use PDOException;
+
+use function array_reduce;
+use function assert;
+use function func_get_args;
+use function is_array;
+
+use const PHP_VERSION_ID;
+
+if (PHP_VERSION_ID >= 80000) {
+    /**
+     * @internal
+     */
+    trait PDOStatementImplementations
+    {
+        /**
+         * {@inheritdoc}
+         *
+         * @deprecated Use one of the fetch- or iterate-related methods.
+         */
+        public function setFetchMode($mode, ...$args)
```

Comment:
Should it be `int $mode` as [documented](https://www.php.net/manual/en/pdostatement.setfetchmode.php)?

Reply:
Same problem, I needed to find a signature that is compatible to the parent PDO class as well as the implemented DBAL interface.

If we wanted `int $mode`, I would need to add the `int` type declaration to `Doctrine\DBAL\Driver\ResultStatement::setFetchMode()` as well.

Reply:
This is by the way something we can safely do. I was just reluctant to change the existing interfaces. I'll let you decide which way to go.

Reply:
It sounds safe but since it's not required, I'm fine with the current state. Thanks for the explanation.